quiche/src/recovery/gcongestion/recovery.rs

```diff
@@ -368,15 +385,33 @@ impl LossThreshold {
     }
 
     fn time_thresh(&self) -> f64 {
-        self.time_thresh
+        match self.time_thresh_overhead {
+            Some(time_thresh_overhead) => 1.0 + time_thresh_overhead,
+            None => self.time_thresh,
+        }
     }
 
     fn on_spurious_loss(&mut self, new_pkt_thresh: u64) {
-        self.pkt_thresh = self
-            .pkt_thresh
-            .max(new_pkt_thresh.min(MAX_PACKET_THRESHOLD));
-
-        self.time_thresh = PACKET_REORDER_TIME_THRESHOLD;
+        match &mut self.time_thresh_overhead {
+            Some(time_thresh_overhead) => {
+                if self.pkt_thresh != DISABLE_PKT_THRESHOLD {
+                    // Disable packet threshold on first spurious loss.
+                    self.pkt_thresh = DISABLE_PKT_THRESHOLD;
+                } else {
```

Comment:
Should the code below always run?

Seems like the `self.time_thresh_overhead.is_none()` branch changes self.time_thresh on the first reorder, but this version waits until the second reorder.